internal/controllers/topology/cluster/desired_state_test.go

```diff
@@ -1828,45 +1850,55 @@ func TestComputeMachineDeploymentVersion(t *testing.T) {
 	//
 	// A machine deployment is considered upgrading if any of the above conditions
 	// is false.
-	machineDeploymentStable := builder.MachineDeployment("test-namespace", "md-1").
-		WithGeneration(1).
-		WithReplicas(2).
-		WithStatus(clusterv1.MachineDeploymentStatus{
-			ObservedGeneration:  2,
-			Replicas:            2,
-			UpdatedReplicas:     2,
-			AvailableReplicas:   2,
-			ReadyReplicas:       2,
-			UnavailableReplicas: 0,
-		}).
-		Build()
-	machineDeploymentRollingOut := builder.MachineDeployment("test-namespace", "md-2").
-		WithGeneration(1).
-		WithReplicas(2).
-		WithStatus(clusterv1.MachineDeploymentStatus{
-			ObservedGeneration:  2,
-			Replicas:            1,
-			UpdatedReplicas:     1,
-			AvailableReplicas:   1,
-			ReadyReplicas:       1,
-			UnavailableReplicas: 1,
-		}).
-		Build()
+	stableMachineDeployment := func(ns, name string) *clusterv1.MachineDeployment {
+		return builder.MachineDeployment(ns, name).
+			WithGeneration(1).
+			WithReplicas(2).
+			WithStatus(clusterv1.MachineDeploymentStatus{
+				ObservedGeneration:  2,
+				Replicas:            2,
+				UpdatedReplicas:     2,
+				AvailableReplicas:   2,
+				ReadyReplicas:       2,
+				UnavailableReplicas: 0,
+			}).
+			Build()
+	}
+
+	rollingMachineDeployment := func(ns, name string) *clusterv1.MachineDeployment {
+		return builder.MachineDeployment(ns, name).
+			WithGeneration(1).
+			WithReplicas(2).
+			WithStatus(clusterv1.MachineDeploymentStatus{
+				ObservedGeneration:  2,
+				Replicas:            1,
+				UpdatedReplicas:     1,
+				AvailableReplicas:   1,
+				ReadyReplicas:       1,
+				UnavailableReplicas: 1,
+			}).
+			Build()
+	}
 
 	machineDeploymentsStateStable := scope.MachineDeploymentsStateMap{
```

Comment:
```suggestion
	twoStableMachineDeploymentsState := scope.MachineDeploymentsStateMap{
```
nit: for consistency with the other two